models/base_model.py

```diff
@@ -336,6 +425,34 @@ def forward(self):
                 self.mask_context, size=self.real_A.shape[2:]
             )[:, 0]
 
+        if any("temporal" in D_name for D_name in self.opt.D_netDs):
+            self.temporal_fake_B = []
+            for i in range(self.opt.D_temporal_number_frames):
```

Comment:
Put this into a function instead of repeating the code for G_A and G_B ?

Reply:
yes!